File: binary_search.cpp
//Binary Search. Call this function in main() to use.

#include <bits/stdc++.h>
using namespace std;
//num is the number to find in the array. l and r are left most and right most element of array.
int binarySearch(int arr[], int l, int r, int num)
{
	if (r >= l) {
		int mid = l + (r - l) / 2;

		// If the element is in middle already
		if (arr[mid] == num)
			return mid;

		// If element is smaller than x then check left subarray
		if (arr[mid] > num)
			return binarySearch(arr, l, mid - 1, num);

		// Else check the right subarray
		return binarySearch(arr, mid + 1, r, num);
	}

	// If element is not present return -1
	return -1;
}

File: Binary-Search/binary search.php
<?php
// PHP program to implement
// recursive Binary Search  
function binarySearch($arr, $l, $r, $x)
{
if ($r >= $l)
{
        $mid = ceil($l + ($r - $l) / 2);
  
        
        if ($arr[$mid] == $x) 
            return floor($mid);
  
        
        if ($arr[$mid] > $x) 
            return binarySearch($arr, $l, 
                                $mid - 1, $x);
  
        
        return binarySearch($arr, $mid + 1, 
                            $r, $x);
}
  

return -1;
}
  

$arr = array(2, 3, 4, 10, 40);
$n = count($arr);
$x = 10;
$result = binarySearch($arr, 0, $n - 1, $x);
if(($result == -1))
echo "Element is not present in array";
else
echo "Element is present at index ",
                            $result;
                              

?>
